paasta_tools/cleanup_kubernetes_jobs.py

```diff
@@ -95,44 +104,118 @@ def alert_state_change(application: Application, soa_dir: str) -> Generator:
         raise
 
 
+def instance_is_not_bouncing(
+    instance_config: KubernetesDeploymentConfig,
+    applications: List[Application],
+    kube_client: KubeClient,
+) -> Optional[bool]:
+    for application in applications:
+        if isinstance(application, DeploymentWrapper):
+            existing_app = application.get_existing_app(kube_client)
+            if (
+                application.kube_deployment.namespace == instance_config.get_namespace()
+                and (
+                    instance_config.config_dict.get("instances", None)
+                    == existing_app.status.ready_replicas
+                    or instance_config.config_dict.get("min_instances", None)
+                    <= existing_app.status.ready_replicas
+                )
```

Comment:
I think I'd use `instance_config.get_instances()` here instead of pulling from the config dicts -- that way you'd get the current HPA numbers.